### PR TITLE
Dockerfile that will run PX4 Flight Review

### PR DESCRIPTION
**Pull request by alhicks**
Usage instructions in base of file - once running access via web-browser using docker instance IP.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+RUN echo PATH="$HOME/.local/bin:$PATH" >> ~/.bashrc
+RUN /bin/bash -c "source ~/.bashrc"
+
+RUN git clone https://github.com/PX4/flight_review.git
+
+WORKDIR "flight_review"
+
+## update caesium and mabox keys
+RUN sed -i "/cesium_api_key =/ s/=.*/=$CESIUM_API_KEY/" config_default.ini
```
Comment (bkueng) [inline Dockerfile]: Can we create a `config_user.ini` for these?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+RUN pip3 install simplekml
+RUN pip3 install matplotlib
+RUN pip3 install smopy
+RUN pip3 install scipy
+RUN pip3 install pyfftw
+
+## USAGE
+## ## ## ## ## ## ## ## ## ## ## ## 
+## BUILD# docker build --build-arg CESIUM_API_KEY=[cesium-key] --build-arg MAPBOX_API_ACCESS_TOKEN=[mapbox_key]  -t px4flightreview .
+## RUN# docker run -it -p 80:5006 px4flightreview python3 serve.py
```
Comment (bkueng) [inline Dockerfile]: Can you add a `CMD` to specify the default startup command?
`pipenv run python serve.py`

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+RUN pip3 install --system pipenv
+
+RUN echo PATH="$HOME/.local/bin:$PATH" >> ~/.bashrc
+RUN /bin/bash -c "source ~/.bashrc"
+
+RUN git clone https://github.com/PX4/flight_review.git
+
+WORKDIR "flight_review"
+
+## update caesium and mabox keys
```
Comment (bkueng) [inline Dockerfile]: Typo:
```suggestion
## update cesium and mabox keys
```

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+RUN pipenv lock --requirements > requirements.txt
+
+## app setup
+RUN pipenv run python setup_db.py
+
+## install bokeh webserver
+RUN pip3 install bokeh
+
+## application required libs
+RUN pip3 install pyulog
```
Comment (bkueng) [inline Dockerfile]: These are not needed as pipenv already installs the deps

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+RUN sed -i "/cesium_api_key =/ s/=.*/=$CESIUM_API_KEY/" config_default.ini
+RUN sed -i "/mapbox_api_access_token =/ s/=.*/=$MAPBOX_API_ACCESS_TOKEN/" config_default.ini
+
+## required for pipenv to run
+ENV LC_ALL=C.UTF-8
+ENV LANG=C.UTF-8
+
+RUN pipenv --three
+RUN pipenv sync
+RUN pipenv lock --requirements > requirements.txt
```
Comment (bkueng) [inline Dockerfile]: You don't need this.

File: Dockerfile (hunk 1 above)
Comment (alhicks) [reply Dockerfile]: 👍🏻

File: Dockerfile (hunk 2 above)
Comment (alhicks) [reply Dockerfile]: 👍🏻

File: Dockerfile (hunk 3 above)
Comment (alhicks) [reply Dockerfile]: 👍🏻

File: Dockerfile (hunk 5 above)
Comment (alhicks) [reply Dockerfile]: 👍🏻

File: Dockerfile (hunk 4 above)
Comment (alhicks) [reply Dockerfile]: Removing the `pip3 install` commands causes failures resolving dependancies so have left in.

Comment (bkueng) [reply Dockerfile]: It works if you keep
```
RUN pipenv --three
RUN pipenv sync
```
And then change the startup command to:
```
CMD ["pipenv", "run", "python3", "serve.py"]
```

Comment (alhicks) [reply Dockerfile]: 👍🏻

**Comment by JohannesBrand** [comment]
@alhicks would it be possible to provide an additional command line script that builds and runs the dockerfile?

and is it necessary to clone the repository in the dockerfile? As I assume that people have already cloned the repository when they build it we could also just copy the local files over, right?

**Comment by alhicks** [comment]
@JohannesBrand - is this so that uses can run their local modified code within the docker container?

**Comment by JohannesBrand** [comment]
@alhicks yes, exactly. Because that's what you usually want, right?

**Comment by alhicks** [comment]
@JohannesBrand my usecase was to run PX4 Flight Review as is from the master branch but I can see how it would be also useful to run from cloned source. Perhaps we should move the current dockerfile to a docker folder off route and keep it for running the master branch version. Then create a new dockerfile that can be used to run the source from the current folder - i.e. local development dockerfile.

thoughts?

**Comment by JohannesBrand** [comment]
@alhicks in my opinion changes should always be made on a feature branch which means your use-case can be met by building the docker container from the master branch (assuming it's clean or changes are stashed) or otherwise it's also easy to keep a second clean version.

I think it could be confusing for developers if this docker container actually runs the latest stable version instead of the local version with changes.

**Comment by alhicks** [comment]
@JohannesBrand good point. I've made the [change](https://github.com/alhicks/flight_review/commit/a3c1732b354197aa91fd9508e4c2459132116265). Does this require a new pull request?

**Comment by bkueng** [comment]
> Does this require a new pull request?

Yes

**Pull request merged**